### PR TITLE
Add max python

### LINKED ISSUE #797: ImageData has no method max
CIL's  [MixedL21Norm convex conjugate](https://github.com/vais-ral/CCPi-Framework/blob/master/Wrappers/Python/ccpi/optimisation/functions/MixedL21Norm.py#L77) expects `x.max()` to give the element-wise maximum of `x`, but if `x` is of ImageData type, it has no method max.

### PR DESCRIPTION
**Pull request by paskino**
closes #797 
addresses #798 for [addition ](github.com/SyneRBI/SIRF/blob/add_max_python/src/common/SIRF.py#L174)and [subtraction](https://github.com/SyneRBI/SIRF/blob/add_max_python/src/common/SIRF.py#L274) and [multiplication](https://github.com/SyneRBI/SIRF/blob/add_max_python/src/common/SIRF.py#L302).

Unfortunately the implementation is different in each method and should be harmonised.

Unittest are currently available as part of [CIL](https://github.com/vais-ral/CCPi-Framework/blob/test_SIRF/Wrappers/Python/test/test_SIRF.py#L150L201) but they really should be part of SIRF.

### CODE REVIEW

**Comment by KrisThielemans** [comment]
that seems to have gone wrong:
```
  File "/home/travis/build/SyneRBI/SIRF/src/xSTIR/pSTIR/tests/tests_four.py", line 40, in test_main
    diff = new_acq_data - acq_data
TypeError: unsupported operand type(s) for -: 'AcquisitionData' and 'AcquisitionData'
```

File: src/common/SIRF.py [hunk 1]
```diff
@@ -171,6 +171,8 @@ def add(self, other, out=None):
         other: DataContainer
         out:   DataContainer to store the result to.
         '''
+        if not isinstance (other, ( DataContainer , Number, int, float, numpy.float32 )):
```
Comment (evgueni-ovtchinnikov) [inline src/common/SIRF.py:174]: int, float and numpy.float32 are derived from Number, are they not?

File: src/common/SIRF.py [hunk 2]
```diff
@@ -267,7 +269,11 @@ def __sub__(self, other):
         data viewed as vectors.
         other: DataContainer
         '''
-        return self.subtract(other)
+        assert self.handle is not None
+
+        if isinstance(other, (Number, int, float, numpy.float32) ):
```
Comment (evgueni-ovtchinnikov) [inline src/common/SIRF.py:274]: DataContainer is missing

**Review by evgueni-ovtchinnikov — CHANGES_REQUESTED** [review]

File: src/common/SIRF.py (hunk 1 above)
Comment (paskino) [reply src/common/SIRF.py:174]: I suppose `int` and `float` may be, but I wouldn't be sure about `numpy.float32`.

Comment (evgueni-ovtchinnikov) [reply src/common/SIRF.py:174]: >>> import numpy
>>> a = numpy.ones((3,), dtype=numpy.float32)
>>> from numbers import Number
>>> print(isinstance(a[0], Number))
True

Comment (paskino) [reply src/common/SIRF.py:174]: very well then

**Comment by paskino** [comment]
@evgueni-ovtchinnikov once travis is happy I will merge, OK?

**Comment by evgueni-ovtchinnikov** [comment]
ok

**Comment by paskino** [comment]
@KrisThielemans @evgueni-ovtchinnikov the build was successful, only the last item in the travis list didn't succeed. It didn't finish the tests, however it built fine. 

Shall I merge anyhow?

**Comment by evgueni-ovtchinnikov** [comment]
I vote you merge

**Comment by KrisThielemans** [comment]
sure

**Pull request merged**